### PR TITLE
fix(schedule): optimize the logic for creating scheduled tasks

### PR DESCRIPTION
**Pull request by guowl3**
#### What type of PR is this?
type-bug

#### What this PR does / why we need it:
1. Refresh the status of task units when restarting a task
2. Optimize the logic for creating scheduled tasks
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskType.java [hunk 1]
```diff
@@ -38,6 +42,29 @@ public enum ScheduleTaskType {
 
     LOGICAL_DATABASE_CHANGE,
 
-    LOAD_DATA
+    LOAD_DATA;
+
+    public boolean isExecuteInTaskFramework() {
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/model/ScheduleTaskType.java:47]: if we can avoid this switch case, cause when new job type added, we will always search the code, and find where the code should be add and which value should be returned. If code not added here, default value returned, and code may not provided any waring. 
Restraint code for user should be designed, whether annotation or interface, make problem found on compile or runtime period.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/DataArchiveDeleteJob.java [hunk 2]
```diff
@@ -37,42 +36,39 @@
 public class DataArchiveDeleteJob extends AbstractDlmJob {
     @Override
     public void executeJob(JobExecutionContext context) {
+        Long scheduleId = ScheduleTaskUtils.getScheduleId(context);
+        Long scheduleTaskId = ScheduleTaskUtils.getScheduleTaskId(context);
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/DataArchiveDeleteJob.java:40]: Control logic can be detached from job, cause they are similar code.

**Review by LioRoger — COMMENTED** [review]
any test added?

**Comment by guowl3** [comment]
move to next version

**Pull request closed without merging**